cmake/Rust.cmake

```diff
@@ -1,52 +1,25 @@
-if(EXISTS "${CMAKE_SOURCE_DIR}/corrosion-vendor/")
-    add_subdirectory("${CMAKE_SOURCE_DIR}/corrosion-vendor/")
-else()
-    include(FetchContent)
-
-    # Don't let Corrosion's tests interfere with ours.
-    set(CORROSION_TESTS OFF CACHE BOOL "" FORCE)
+# Trying to build using the resolved toolchain causes all kinds of weird errors
+# Just let rustup do its job
+set(Rust_RESOLVE_RUSTUP_TOOLCHAINS Off)
 
-    FetchContent_Declare(
-        Corrosion
-        GIT_REPOSITORY https://github.com/mqudsi/corrosion
-        GIT_TAG fish
-    )
+include(FindRust)
+find_package(Rust REQUIRED)
```

Comment:
We could make this check the version:

```cmake
find_package(Rust 1.67 REQUIRED)
```

As I understand with only a single version that's a lower bound, and equal or later versions are accepted.